src/test/java/stepDefs/DefinitionsSteps.java

```diff
@@ -0,0 +1,103 @@
+package stepDefs;
+
+import static constants.Constants.BOOK_PRODUCT_URL;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import desktop.pageFactory.PageFactoryManager;
+import io.cucumber.java.After;
+import io.cucumber.java.Before;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.firefox.FirefoxDriver;
+
+import java.util.List;
+import java.util.Map;
+
+public class DefinitionsSteps {
+
+    WebDriver driver;
+    PageFactoryManager pageFactoryManager;
+
+    @Before
+    public void driverSetUp() {
+        String browser = System.getProperty("browser");
+        if (browser.equalsIgnoreCase("firefox")) {
+            System.setProperty("webdriver.gecko.driver", "src/main/resources/geckodriver.exe");
+            driver = new FirefoxDriver();
+        }
+        if (browser.equalsIgnoreCase("chrome")) {
+            System.setProperty("webdriver.chrome.driver", "src/main/resources/chromedriver.exe");
+            driver = new ChromeDriver();
+        }
+        driver.manage().window().maximize();
+        pageFactoryManager = new PageFactoryManager(driver);
+    }
+
+
+    @Given("I open Book Depository")
+    public void openHomePage() {
+        pageFactoryManager.getHomePage().openHomePage();
+    }
+
+    @Given("I open Thinking in Java product")
+    public void openProduct() {
+        driver.get(BOOK_PRODUCT_URL);
+    }
+
+    @Given("I manually accept cookie")
+    public void acceptCookie() {
+        pageFactoryManager.getBasePage().acceptCookie();
+    }
+
+    @When("I apply the following search filters")
+    public void chooseSearchFilters(Map<String, String> filter) {
+        pageFactoryManager.getSearchPage().selectFilter(filter);
+    }
+
+    @When("I search for {string}")
+    public void search(String keyword) {
+        pageFactoryManager.getHomePage().search(keyword);
+    }
+
+    @Then("Search results contain only the following products")
+    public void searchResultsContainOnlyProducts(List<String> bookList) {
+        assertThat(bookList).isEqualTo(pageFactoryManager.getSearchPage().searchResultBooksListText());
```

Comment:
It is also a good practice to add some description to AssertJ (similar to error message in JUnit) using as() method. After that, it would look like this - `assertThat(bookList).as("Checking book list").isEqualTo(pageFactoryManager.getSearchPage().searchResultBooksListText());`